frontend/app_flowy/lib/workspace/presentation/settings/widgets/settings_file_exporter_widget.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:flowy_sdk/dispatch/dispatch.dart';
+import 'package:flowy_sdk/protobuf/flowy-error/errors.pbserver.dart';
+import 'package:flowy_sdk/protobuf/flowy-folder/app.pb.dart';
+import 'package:flowy_sdk/protobuf/flowy-folder/workspace.pb.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../generated/locale_keys.g.dart';
+
+class FileExporterWidget extends StatefulWidget {
+  const FileExporterWidget({Key? key}) : super(key: key);
+
+  @override
+  State<FileExporterWidget> createState() => _FileExporterWidgetState();
+}
+
+class _FileExporterWidgetState extends State<FileExporterWidget> {
+  // Map<String, List<String>> _selectedPages = {};
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        FlowyText.medium(
+          LocaleKeys.settings_files_selectFiles.tr(),
+          fontSize: 16.0,
+        ),
+        const VSpace(8),
+        Expanded(child: _buildFileSelector(context)),
+        const VSpace(8),
+        _buildButtons(context)
+      ],
+    );
+  }
+
+  Row _buildButtons(BuildContext context) {
+    return Row(
+      children: [
+        const Spacer(),
+        FlowyTextButton(
+          LocaleKeys.button_Cancel.tr(),
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+        ),
+        const HSpace(8),
+        FlowyTextButton(
+          LocaleKeys.button_OK.tr(),
+          onPressed: () async {
+            // TODO: Export Data
+            await FilePicker.platform.getDirectoryPath().then((exportPath) {
+              Navigator.of(context).pop();
+            });
+          },
+        ),
+      ],
+    );
+  }
+
+  FutureBuilder<dartz.Either<WorkspaceSettingPB, FlowyError>>
+      _buildFileSelector(BuildContext context) {
+    return FutureBuilder<dartz.Either<WorkspaceSettingPB, FlowyError>>(
+      future: FolderEventReadCurrentWorkspace().send(),
+      builder: (context, snapshot) {
+        if (snapshot.hasData &&
+            snapshot.connectionState == ConnectionState.done) {
+          final workspaces = snapshot.data?.getLeftOrNull<WorkspaceSettingPB>();
+          if (workspaces != null) {
+            return _ExpandedList(
+              apps: workspaces.workspace.apps.items,
+              onChanged: (selectedPages) {
+                // _selectedPages = selectedPages;
+              },
+            );
+          }
+        }
+        return const CircularProgressIndicator();
+      },
+    );
+  }
+}
+
+class _ExpandedList extends StatefulWidget {
+  const _ExpandedList({
+    Key? key,
+    required this.apps,
+    required this.onChanged,
+  }) : super(key: key);
+
+  final List<AppPB> apps;
+  final void Function(Map<String, List<String>> selectedPages) onChanged;
+
+  @override
+  State<_ExpandedList> createState() => __ExpandedListState();
+}
+
+class __ExpandedListState extends State<_ExpandedList> {
```

Comment:
Maybe we could have just one underscore? Or is there some other meaning?

Reply:
Actually, this `state` was auto-generated by VSCode. Like, PublicWidget -> _PrivateState, _PrivateWidget -> __PrivateState.

I agree that the double underscore here is meaningless.